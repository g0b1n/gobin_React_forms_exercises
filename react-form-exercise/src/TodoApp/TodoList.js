import React, {useState} from "react";
import { v4 as uuid } from 'uuid';
import Todo from "./Todo";
import NewTodoForm from "../TodoApp/NewTodoForm";


const TodoList = () => {
    const InitialState = [
        {id: uuid(), todo: "Go for a walk"},
        {id: uuid(), todo: "Hit the gym"}
    ]

    const [todo, setTodo] = useState(InitialState);
    const addTodo = (newTodo) => {
        setTodo(todos => [...todos, {...newTodo, id: uuid() }]);
    }

    // Delete Todo
    const deleteTodo = (id) => {
        setTodo(todos => todos.filter(todo => todo.id !== id));
    }

    return (
        <div className="TodoList">
            <NewTodoForm addTodo={addTodo}/>
            <h3>My Todo List</h3>
                {todo.map(todo => (
                    <div key={todo.id}>
                        <button onClick={ () => deleteTodo(todo.id)}>X</button>
                        <Todo
                            id={todo.id}
                            todo={todo.todo}
                        />
                    </div>
            ))}
        </div>
    )
}

export default TodoList;