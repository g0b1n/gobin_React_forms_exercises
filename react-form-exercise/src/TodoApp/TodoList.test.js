import React from "react";
import { render } from "@testing-library/react";
import TodoList from "./TodoList";

// smoke test
test('it renders without crashing', () => {
    render(<TodoList />)
})

// snapshot test
test('it matches the snapshot', () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot()
})