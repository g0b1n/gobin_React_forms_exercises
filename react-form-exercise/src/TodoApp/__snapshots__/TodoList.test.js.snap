// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <form>
      <label
        for="todo"
      >
        Todo
      </label>
      <input
        id="todo"
        name="todo"
        placeholder="Add new todo"
        type="text"
        value=""
      />
      <button>
        Add New Todo
      </button>
    </form>
    <h3>
      My Todo List
    </h3>
    <div>
      <button>
        X
      </button>
      <ul>
        <li>
          Go for a walk
        </li>
      </ul>
    </div>
    <div>
      <button>
        X
      </button>
      <ul>
        <li>
          Hit the gym
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
